uber/site_sections/preregistration.py

```diff
@@ -524,3 +527,22 @@ def shirt_reorder(self, session, message='', **params):
             'message': message,
             'attendee': attendee
         }
+
+    def apply_promo_code(self, session, message='', **params):
+        if 'id' and 'code' in params:
+            if params['code'] is not '':
+                attendee, group = self._get_unsaved(params['id'])
+                if not attendee:
+                    message = 'Attendee Not Found'
+                else:
+                    promo_code = session.query(PromoCode).filter(PromoCode.code == params['code'] and not PromoCode.expired).first()
+                    if not promo_code:
+                        message = "Promo Code Not Found"
+                    else:
+                        if attendee.promo_code_id is not None:
+                            message = "Promo Code Already Being Used"
+                        else:
+                            attendee.promo_code_id = promo_code.id
```

Comment:
In light of the relationship change, this will be `attendee.promo_code = promo_code`. I think.